h/static/scripts/controllers/search-bar-controller.js

```diff
@@ -0,0 +1,139 @@
+'use strict';
+
+var Controller = require('../base/controller');
+var setElementState = require('../util/dom').setElementState;
+
+/**
+ * Controller for the search bar.
+ */
+class SearchBarController extends Controller {
+  constructor(element) {
+    super(element);
+
+    this._input = element.querySelector('.js-search-bar-input');
+    this._dropdown = element.querySelector('.js-search-bar-dropdown');
+    this._dropdownItems = element.querySelectorAll('.js-search-bar-dropdown-item');
+
+    var self = this;
+    var index = -1;
+    var menuLen = this._dropdownItems.length;
```

Comment:
I think you should just access this as `this._dropdownItems.length` when you need it, rather than saving it in a class-level variable. Removes an unnecessary variable, but also it should continue working when, in future versions of this menu, the number of dropdown items can change dynamically